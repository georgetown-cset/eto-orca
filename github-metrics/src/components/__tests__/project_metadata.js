import React from "react"
import {render} from "@testing-library/react"
import ProjectMetadata from "../project_metadata"

describe("Project metadata", () => {
  it("renders correctly with no references", () => {
    const card = render(
      <ProjectMetadata
        data={{
          "stargazers_count": 123,
          "num_contributors": 45,
          "open_issues": 56,
          "created_at": "2023-01-01",
          "pushed_at": "2023-03-14",
          "license": "MIT",
          "language": "Python"
        }}
      />
    );
    expect(card).toMatchSnapshot();
    expect(card.queryAllByText("Mentions")).toHaveLength(0);
  });

  it("renders correctly with references", () => {
    const card = render(
      <ProjectMetadata
        field={"Algebraic Geometry"}
        showNumReferences={true}
        data={{
          "stargazers_count": 123,
          "num_contributors": 45,
          "open_issues": 56,
          "created_at": "2023-01-01",
          "pushed_at": "2023-03-14",
          "license": "MIT",
          "language": "Python",
          "num_references": {"Algebraic Geometry": 76},
          "relevance": {"Algebraic Geometry": 42.42}
        }}
      />
    );
    expect(card).toMatchSnapshot();
    expect(card.queryAllByText("mentions", {exact: false})).toHaveLength(1);
  });

});
