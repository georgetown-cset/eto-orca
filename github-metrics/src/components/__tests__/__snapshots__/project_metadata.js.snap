// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project metadata renders correctly with no references 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="MuiTypography-root MuiTypography-body2 css-rqgorj-MuiTypography-root-sortOption"
        >
          <span
            class="css-mmmexm-metaSection"
          >
            <span
              class="css-1kzx35s-emph"
            >
              Stars
            </span>
            : 
            123
          </span>
          <span
            class="css-mmmexm-metaSection"
          >
            <span
              class="css-1kzx35s-emph"
            >
              Watchers
            </span>
            : 
            0
          </span>
        </div>
        <div
          class="MuiTypography-root MuiTypography-body2 css-rqgorj-MuiTypography-root-sortOption"
        >
          <span
            class="css-mmmexm-metaSection"
          >
            <span
              class="css-1kzx35s-emph"
            >
              Open Issues
            </span>
            : 
            56
          </span>
          <span
            class="css-mmmexm-metaSection"
          >
            <span
              class="css-1kzx35s-emph"
            >
              Contributors
            </span>
            : 
            45
          </span>
        </div>
        <div
          class="MuiTypography-root MuiTypography-body2 css-rqgorj-MuiTypography-root-sortOption"
        >
          <span
            class="css-mmmexm-metaSection"
          >
            <span
              class="css-1kzx35s-emph"
            >
              Created
            </span>
            : 
            2023-01-01
          </span>
          <span
            class="css-mmmexm-metaSection"
          >
            <span
              class="css-1kzx35s-emph"
            >
              Last Commit
            </span>
            : 
            2023-03-14
          </span>
        </div>
        <div
          class="MuiTypography-root MuiTypography-body2 css-rqgorj-MuiTypography-root-sortOption"
        >
          <span
            class="css-mmmexm-metaSection"
          >
            <span
              class="css-1kzx35s-emph"
            >
              License
            </span>
            : 
            MIT
          </span>
        </div>
        <div
          class="MuiTypography-root MuiTypography-body2 css-rqgorj-MuiTypography-root-sortOption"
        >
          <span
            class="css-mmmexm-metaSection"
          >
            <span
              class="css-1kzx35s-emph"
            >
              Top Programming Language
            </span>
            : 
            Python
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="MuiTypography-root MuiTypography-body2 css-rqgorj-MuiTypography-root-sortOption"
      >
        <span
          class="css-mmmexm-metaSection"
        >
          <span
            class="css-1kzx35s-emph"
          >
            Stars
          </span>
          : 
          123
        </span>
        <span
          class="css-mmmexm-metaSection"
        >
          <span
            class="css-1kzx35s-emph"
          >
            Watchers
          </span>
          : 
          0
        </span>
      </div>
      <div
        class="MuiTypography-root MuiTypography-body2 css-rqgorj-MuiTypography-root-sortOption"
      >
        <span
          class="css-mmmexm-metaSection"
        >
          <span
            class="css-1kzx35s-emph"
          >
            Open Issues
          </span>
          : 
          56
        </span>
        <span
          class="css-mmmexm-metaSection"
        >
          <span
            class="css-1kzx35s-emph"
          >
            Contributors
          </span>
          : 
          45
        </span>
      </div>
      <div
        class="MuiTypography-root MuiTypography-body2 css-rqgorj-MuiTypography-root-sortOption"
      >
        <span
          class="css-mmmexm-metaSection"
        >
          <span
            class="css-1kzx35s-emph"
          >
            Created
          </span>
          : 
          2023-01-01
        </span>
        <span
          class="css-mmmexm-metaSection"
        >
          <span
            class="css-1kzx35s-emph"
          >
            Last Commit
          </span>
          : 
          2023-03-14
        </span>
      </div>
      <div
        class="MuiTypography-root MuiTypography-body2 css-rqgorj-MuiTypography-root-sortOption"
      >
        <span
          class="css-mmmexm-metaSection"
        >
          <span
            class="css-1kzx35s-emph"
          >
            License
          </span>
          : 
          MIT
        </span>
      </div>
      <div
        class="MuiTypography-root MuiTypography-body2 css-rqgorj-MuiTypography-root-sortOption"
      >
        <span
          class="css-mmmexm-metaSection"
        >
          <span
            class="css-1kzx35s-emph"
          >
            Top Programming Language
          </span>
          : 
          Python
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Project metadata renders correctly with references 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="MuiTypography-root MuiTypography-body2 css-rqgorj-MuiTypography-root-sortOption"
        >
          <span
            class="css-mmmexm-metaSection"
          >
            <span
              class="css-1kzx35s-emph"
            >
              Stars
            </span>
            : 
            123
          </span>
          <span
            class="css-mmmexm-metaSection"
          >
            <span
              class="css-1kzx35s-emph"
            >
              Watchers
            </span>
            : 
            0
          </span>
          <span
            class="css-mmmexm-metaSection"
          >
            <span
              class="css-1kzx35s-emph"
            >
              References
            </span>
            : 
            76
          </span>
        </div>
        <div
          class="MuiTypography-root MuiTypography-body2 css-rqgorj-MuiTypography-root-sortOption"
        >
          <span
            class="css-mmmexm-metaSection"
          >
            <span
              class="css-1kzx35s-emph"
            >
              Open Issues
            </span>
            : 
            56
          </span>
          <span
            class="css-mmmexm-metaSection"
          >
            <span
              class="css-1kzx35s-emph"
            >
              Contributors
            </span>
            : 
            45
          </span>
        </div>
        <div
          class="MuiTypography-root MuiTypography-body2 css-rqgorj-MuiTypography-root-sortOption"
        >
          <span
            class="css-mmmexm-metaSection"
          >
            <span
              class="css-1kzx35s-emph"
            >
              Created
            </span>
            : 
            2023-01-01
          </span>
          <span
            class="css-mmmexm-metaSection"
          >
            <span
              class="css-1kzx35s-emph"
            >
              Last Commit
            </span>
            : 
            2023-03-14
          </span>
        </div>
        <div
          class="MuiTypography-root MuiTypography-body2 css-rqgorj-MuiTypography-root-sortOption"
        >
          <span
            class="css-mmmexm-metaSection"
          >
            <span
              class="css-1kzx35s-emph"
            >
              License
            </span>
            : 
            MIT
          </span>
        </div>
        <div
          class="MuiTypography-root MuiTypography-body2 css-rqgorj-MuiTypography-root-sortOption"
        >
          <span
            class="css-mmmexm-metaSection"
          >
            <span
              class="css-1kzx35s-emph"
            >
              Top Programming Language
            </span>
            : 
            Python
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="MuiTypography-root MuiTypography-body2 css-rqgorj-MuiTypography-root-sortOption"
      >
        <span
          class="css-mmmexm-metaSection"
        >
          <span
            class="css-1kzx35s-emph"
          >
            Stars
          </span>
          : 
          123
        </span>
        <span
          class="css-mmmexm-metaSection"
        >
          <span
            class="css-1kzx35s-emph"
          >
            Watchers
          </span>
          : 
          0
        </span>
        <span
          class="css-mmmexm-metaSection"
        >
          <span
            class="css-1kzx35s-emph"
          >
            References
          </span>
          : 
          76
        </span>
      </div>
      <div
        class="MuiTypography-root MuiTypography-body2 css-rqgorj-MuiTypography-root-sortOption"
      >
        <span
          class="css-mmmexm-metaSection"
        >
          <span
            class="css-1kzx35s-emph"
          >
            Open Issues
          </span>
          : 
          56
        </span>
        <span
          class="css-mmmexm-metaSection"
        >
          <span
            class="css-1kzx35s-emph"
          >
            Contributors
          </span>
          : 
          45
        </span>
      </div>
      <div
        class="MuiTypography-root MuiTypography-body2 css-rqgorj-MuiTypography-root-sortOption"
      >
        <span
          class="css-mmmexm-metaSection"
        >
          <span
            class="css-1kzx35s-emph"
          >
            Created
          </span>
          : 
          2023-01-01
        </span>
        <span
          class="css-mmmexm-metaSection"
        >
          <span
            class="css-1kzx35s-emph"
          >
            Last Commit
          </span>
          : 
          2023-03-14
        </span>
      </div>
      <div
        class="MuiTypography-root MuiTypography-body2 css-rqgorj-MuiTypography-root-sortOption"
      >
        <span
          class="css-mmmexm-metaSection"
        >
          <span
            class="css-1kzx35s-emph"
          >
            License
          </span>
          : 
          MIT
        </span>
      </div>
      <div
        class="MuiTypography-root MuiTypography-body2 css-rqgorj-MuiTypography-root-sortOption"
      >
        <span
          class="css-mmmexm-metaSection"
        >
          <span
            class="css-1kzx35s-emph"
          >
            Top Programming Language
          </span>
          : 
          Python
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
